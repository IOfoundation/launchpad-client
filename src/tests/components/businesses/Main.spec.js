import React from 'react';
import {shallow} from 'enzyme';
import {expect} from 'chai';
import BusinessesView from '../../../components/businesses/Main';

const businesses = {
  locations: [],
  organizations: [
    {id: 1, name: 'organization 1'},
    {id: 2, name: 'organization 2'},
  ],
  items: [],
  metadata: {},
  filters: {
    businessServices: [],
    businessTypes: [(0: {}), (1: {}), (2: {})],
    stages: [],
    communities: [],
  },
  displayOptions: {
    showBusinessTypes: true,
    locationToggleSwitch: false,
  },
};

const isMobile = false;
const showLoading = false;

const handleOnChangeBusinessType = jest.fn();
const handleOnChangeLocationToggle = jest.fn();
const handleChangePage = jest.fn();
const handleClickOnClearAllFilters = jest.fn();
const handleOnChangeFilterOptions = jest.fn();

describe('<BusinessesView />', () => {
  it('Renders a ContentMap component', () => {
    const wrapper = shallow(
      <BusinessesView
        isMobile={isMobile}
        businesses={businesses}
        showLoading={showLoading}
        handleOnChangeBusinessType={handleOnChangeBusinessType}
        handleOnChangeLocationToggle={handleOnChangeLocationToggle}
        handleChangePage={handleChangePage}
        handleClickOnClearAllFilters={handleClickOnClearAllFilters}
        handleOnChangeFilterOptions={handleOnChangeFilterOptions}
      />
    );
    expect(wrapper.find('ContentMap').length).to.equal(1);
  });

  it('Renders a BusinessesList component', () => {
    const wrapper = shallow(
      <BusinessesView
        isMobile={isMobile}
        businesses={businesses}
        showLoading={showLoading}
        handleOnChangeBusinessType={handleOnChangeBusinessType}
        handleOnChangeLocationToggle={handleOnChangeLocationToggle}
        handleChangePage={handleChangePage}
        handleClickOnClearAllFilters={handleClickOnClearAllFilters}
        handleOnChangeFilterOptions={handleOnChangeFilterOptions}
      />
    );
    expect(wrapper.find('BusinessesList').length).to.equal(1);
  });

  it('Renders a Pagination component', () => {
    const wrapper = shallow(
      <BusinessesView
        isMobile={isMobile}
        businesses={businesses}
        showLoading={showLoading}
        handleOnChangeBusinessType={handleOnChangeBusinessType}
        handleOnChangeLocationToggle={handleOnChangeLocationToggle}
        handleChangePage={handleChangePage}
        handleClickOnClearAllFilters={handleClickOnClearAllFilters}
        handleOnChangeFilterOptions={handleOnChangeFilterOptions}
      />
    );
    expect(wrapper.find('Pagination').length).to.equal(1);
  });

  it('Renders resultsInfo component', () => {
    const wrapper = shallow(
      <BusinessesView
        isMobile={isMobile}
        businesses={businesses}
        showLoading={showLoading}
        handleOnChangeBusinessType={handleOnChangeBusinessType}
        handleOnChangeLocationToggle={handleOnChangeLocationToggle}
        handleChangePage={handleChangePage}
        handleClickOnClearAllFilters={handleClickOnClearAllFilters}
        handleOnChangeFilterOptions={handleOnChangeFilterOptions}
      />
    );
    expect(wrapper.find('ContentMap').exists('ResultsInfo')).to.equal(true);
  });
});
