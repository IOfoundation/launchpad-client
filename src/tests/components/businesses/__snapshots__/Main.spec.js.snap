// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> renders snapshot of Main 1`] = `
<ContentMap
  businesses={
    Object {
      "appliedFilters": Object {
        "page": 1,
      },
      "displayOptions": Object {
        "locationToggleSwitch": false,
        "showBusinessTypes": true,
      },
      "filters": Object {
        "businessServices": Array [
          Object {
            "children": Array [
              Object {
                "id": 3,
                "name": "Bookkeeping/Accounting Software Training",
              },
              Object {
                "id": 4,
                "name": "Budget, Financial Statement & Cash Flow Training",
              },
            ],
            "id": 2,
            "name": "Financial Management",
          },
          Object {
            "children": Array [
              Object {
                "id": 11,
                "name": "Accelerator Programs",
              },
              Object {
                "id": 12,
                "name": "Equity Investors: Angels",
              },
            ],
            "id": 10,
            "name": "Capital",
          },
        ],
        "businessTypes": Array [
          Object {
            "id": 142,
            "name": "Startup or High-Growth Business",
          },
          Object {
            "id": 143,
            "name": "Main Street or Small Business",
          },
          Object {
            "id": 144,
            "name": "Microenterprise or Home Based Business",
          },
        ],
        "communities": Array [
          Object {
            "id": 151,
            "name": "African American",
          },
          Object {
            "id": 156,
            "name": "Native American",
          },
        ],
        "industries": Array [
          Object {
            "id": 172,
            "name": "Management",
          },
          Object {
            "id": 180,
            "name": "Retail",
          },
        ],
        "stages": Array [
          Object {
            "id": 146,
            "name": "Idea/Inception",
          },
          Object {
            "id": 147,
            "name": "Proof of Concept/Prototype/Market Intro",
          },
        ],
      },
      "items": Array [],
      "locations": Array [],
      "metadata": Object {},
      "organizations": Array [
        Object {
          "contacts": Array [
            Object {
              "email": "jondoe@test.org",
              "id": 1,
              "name": "Jon Doe",
              "phones": Array [
                Object {
                  "number": "(530) 795-1520",
                },
              ],
            },
          ],
          "description": "orgDescription",
          "email": "jondoe@something.org",
          "id": 1,
          "locations": Array [
            Object {
              "address": Object {
                "address_1": "123 Road St",
                "address_2": null,
                "city": "Sacramento",
                "postal_code": "12345",
                "state_province": "CA",
              },
              "id": 1,
              "name": "location1",
              "phones": Array [
                Object {
                  "id": 2,
                  "number": "(123) 456 - 7890",
                },
              ],
            },
            Object {
              "address": Object {
                "address_1": "456 Street Rd",
                "address_2": null,
                "city": "Sacramento",
                "postal_code": "12345",
                "state_province": "CA",
              },
              "id": 2,
              "name": "location2",
              "phones": Array [
                Object {
                  "id": 3,
                  "number": "(123) 456 - 7890",
                },
              ],
            },
          ],
          "name": "organization1",
          "phones": Array [
            Object {
              "id": 1,
              "number": "(123) 456 - 7890",
            },
          ],
          "services": Array [
            Object {
              "categories": Array [
                Object {
                  "id": 54,
                  "name": "category1",
                },
                Object {
                  "id": 55,
                  "name": "category2",
                },
                Object {
                  "id": 56,
                  "name": "category3",
                },
              ],
              "description": "lorem ipsum...",
              "email": null,
              "id": 1,
              "name": "service1",
              "phones": Array [],
            },
            Object {
              "categories": Array [
                Object {
                  "id": 54,
                  "name": "category1",
                },
              ],
              "description": "lorem ipsum...",
              "email": null,
              "id": 2,
              "name": "service2",
              "phones": Array [],
            },
          ],
        },
        Object {
          "contacts": Array [
            Object {
              "email": "jondoe@test.org",
              "id": 1,
              "name": "Jon Doe",
              "phones": Array [
                Object {
                  "number": "(530) 795-1520",
                },
              ],
            },
          ],
          "description": "orgDescription",
          "email": "jondoe@something.org",
          "id": 1,
          "locations": Array [
            Object {
              "address": Object {
                "address_1": "123 Road St",
                "address_2": null,
                "city": "Sacramento",
                "postal_code": "12345",
                "state_province": "CA",
              },
              "id": 1,
              "name": "location1",
              "phones": Array [
                Object {
                  "id": 2,
                  "number": "(123) 456 - 7890",
                },
              ],
            },
            Object {
              "address": Object {
                "address_1": "456 Street Rd",
                "address_2": null,
                "city": "Sacramento",
                "postal_code": "12345",
                "state_province": "CA",
              },
              "id": 2,
              "name": "location2",
              "phones": Array [
                Object {
                  "id": 3,
                  "number": "(123) 456 - 7890",
                },
              ],
            },
          ],
          "name": "organization1",
          "phones": Array [
            Object {
              "id": 1,
              "number": "(123) 456 - 7890",
            },
          ],
          "services": Array [
            Object {
              "categories": Array [
                Object {
                  "id": 54,
                  "name": "category1",
                },
                Object {
                  "id": 55,
                  "name": "category2",
                },
                Object {
                  "id": 56,
                  "name": "category3",
                },
              ],
              "description": "lorem ipsum...",
              "email": null,
              "id": 1,
              "name": "service1",
              "phones": Array [],
            },
            Object {
              "categories": Array [
                Object {
                  "id": 54,
                  "name": "category1",
                },
              ],
              "description": "lorem ipsum...",
              "email": null,
              "id": 2,
              "name": "service2",
              "phones": Array [],
            },
          ],
        },
      ],
    }
  }
  expandMap={[Function]}
  expanded={false}
  handleChangePage={[Function]}
  highlightOrgCard={[Function]}
  isMobile={false}
  onBoundsChange={[Function]}
  redoSearchInMap={[Function]}
  reduceMap={[Function]}
  selectedOrg={-1}
  showLoading={false}
  toggleSwitch={false}
  topBar={
    <ResultInfo
      filterOptions={
        Array [
          Object {
            "id": 142,
            "name": "Startup or High-Growth Business",
          },
          Object {
            "id": 143,
            "name": "Main Street or Small Business",
          },
          Object {
            "id": 144,
            "name": "Microenterprise or Home Based Business",
          },
        ]
      }
      handleOnChangeFilterOptions={[Function]}
      metadata={Object {}}
      showBusinessTypes={true}
      showLoading={false}
    />
  }
>
  <div>
    <BusinessesList
      isMobile={false}
      organizations={
        Array [
          Object {
            "contacts": Array [
              Object {
                "email": "jondoe@test.org",
                "id": 1,
                "name": "Jon Doe",
                "phones": Array [
                  Object {
                    "number": "(530) 795-1520",
                  },
                ],
              },
            ],
            "description": "orgDescription",
            "email": "jondoe@something.org",
            "id": 1,
            "locations": Array [
              Object {
                "address": Object {
                  "address_1": "123 Road St",
                  "address_2": null,
                  "city": "Sacramento",
                  "postal_code": "12345",
                  "state_province": "CA",
                },
                "id": 1,
                "name": "location1",
                "phones": Array [
                  Object {
                    "id": 2,
                    "number": "(123) 456 - 7890",
                  },
                ],
              },
              Object {
                "address": Object {
                  "address_1": "456 Street Rd",
                  "address_2": null,
                  "city": "Sacramento",
                  "postal_code": "12345",
                  "state_province": "CA",
                },
                "id": 2,
                "name": "location2",
                "phones": Array [
                  Object {
                    "id": 3,
                    "number": "(123) 456 - 7890",
                  },
                ],
              },
            ],
            "name": "organization1",
            "phones": Array [
              Object {
                "id": 1,
                "number": "(123) 456 - 7890",
              },
            ],
            "services": Array [
              Object {
                "categories": Array [
                  Object {
                    "id": 54,
                    "name": "category1",
                  },
                  Object {
                    "id": 55,
                    "name": "category2",
                  },
                  Object {
                    "id": 56,
                    "name": "category3",
                  },
                ],
                "description": "lorem ipsum...",
                "email": null,
                "id": 1,
                "name": "service1",
                "phones": Array [],
              },
              Object {
                "categories": Array [
                  Object {
                    "id": 54,
                    "name": "category1",
                  },
                ],
                "description": "lorem ipsum...",
                "email": null,
                "id": 2,
                "name": "service2",
                "phones": Array [],
              },
            ],
          },
          Object {
            "contacts": Array [
              Object {
                "email": "jondoe@test.org",
                "id": 1,
                "name": "Jon Doe",
                "phones": Array [
                  Object {
                    "number": "(530) 795-1520",
                  },
                ],
              },
            ],
            "description": "orgDescription",
            "email": "jondoe@something.org",
            "id": 1,
            "locations": Array [
              Object {
                "address": Object {
                  "address_1": "123 Road St",
                  "address_2": null,
                  "city": "Sacramento",
                  "postal_code": "12345",
                  "state_province": "CA",
                },
                "id": 1,
                "name": "location1",
                "phones": Array [
                  Object {
                    "id": 2,
                    "number": "(123) 456 - 7890",
                  },
                ],
              },
              Object {
                "address": Object {
                  "address_1": "456 Street Rd",
                  "address_2": null,
                  "city": "Sacramento",
                  "postal_code": "12345",
                  "state_province": "CA",
                },
                "id": 2,
                "name": "location2",
                "phones": Array [
                  Object {
                    "id": 3,
                    "number": "(123) 456 - 7890",
                  },
                ],
              },
            ],
            "name": "organization1",
            "phones": Array [
              Object {
                "id": 1,
                "number": "(123) 456 - 7890",
              },
            ],
            "services": Array [
              Object {
                "categories": Array [
                  Object {
                    "id": 54,
                    "name": "category1",
                  },
                  Object {
                    "id": 55,
                    "name": "category2",
                  },
                  Object {
                    "id": 56,
                    "name": "category3",
                  },
                ],
                "description": "lorem ipsum...",
                "email": null,
                "id": 1,
                "name": "service1",
                "phones": Array [],
              },
              Object {
                "categories": Array [
                  Object {
                    "id": 54,
                    "name": "category1",
                  },
                ],
                "description": "lorem ipsum...",
                "email": null,
                "id": 2,
                "name": "service2",
                "phones": Array [],
              },
            ],
          },
        ]
      }
      selectedOrg={-1}
    />
    <Pagination
      appliedFilters={
        Object {
          "page": 1,
        }
      }
      handleChangePage={[Function]}
      metadata={Object {}}
    />
  </div>
</ContentMap>
`;
